src/backend/looped/vehicle/navigation.cpp

```diff
@@ -22,27 +31,270 @@ namespace big
 
 		virtual void on_tick() override
 		{
-			if (g.self.nav_voice)
-			{
-				if (g_player_service->get_self()->get_current_vehicle() != nullptr)
-					g_fiber_pool->queue_job([] {
-						int direction;
-						Vector3 waypoint;
-						float dist;
-						blip::get_blip_location(waypoint, (int)BlipIcons::Waypoint);
-						bool doesPointExist = blip::get_blip_location(waypoint, (int)BlipIcons::Waypoint);
-						float v;
-						PATHFIND::GENERATE_DIRECTIONS_TO_COORD(waypoint.x, waypoint.y, waypoint.z, 1, &direction, &v, &dist);
-
-						auto plyr = g_player_service->get_self();
-						if (doesPointExist)
-							LOG(INFO) << std::format("{}                                              {}", direction, dist);
-
-						if (plyr->get_current_vehicle() != nullptr)
-						{
-						}
-					});
-			}
+			// taken from a native db website.
+			/*
+			distToNxJunction seems to be the distance in metres * 10.0f
+
+		direction:
+
+		0 = This happens randomly during the drive for seemingly no reason but if you consider that this native is only used in trevor3, it seems to mean "Next frame, stop whatever's being said and tell the player the direction."
+		1 = Route is being calculated or the player is going in the wrong direction
+		2 = Please Proceed the Highlighted Route
+		3 = In (distToNxJunction) Turn Left
+		4 = In (distToNxJunction) Turn Right
+		5 = In (distToNxJunction) Keep Straight
+		6 = In (distToNxJunction) Turn Sharply To The Left
+		7 = In (distToNxJunction) Turn Sharply To The Right
+		8 = Route is being recalculated or the navmesh is confusing. This happens randomly during the drive but consistently at {2044.0358, 2996.6116, 44.9717} if you face towards the bar and the route needs you to turn right. In that particular case, it could be a bug with how the turn appears to be 270 deg. CCW instead of "right." Either way, this seems to be the engine saying "I don't know the route right now."
+			*/
+
+
+			g_fiber_pool->queue_job([] {
+				int direction;
+				int feetToAction;
+				Vector3 waypoint;
+				float dist;
+				float v;
+				// most important variables have been declared upfront
+				blip::get_blip_location(waypoint, (int)BlipIcons::Waypoint);
+				bool doesPointExist = blip::get_blip_location(waypoint, (int)BlipIcons::Waypoint);
+				// the if statement allows the function to not run if there is no player waypoint
+				if (doesPointExist)
+				{
+					PATHFIND::GENERATE_DIRECTIONS_TO_COORD(waypoint.x, waypoint.y, waypoint.z, 1, &direction, &v, &dist);
+					feetToAction = math::divided_by_ten(math::meters_to_feet(dist));
```

Comment:
Seriously? You need a helper function to divide by ten?

Reply:
It is to make it more streamlined.  Plus it is inlined so it is pretty much doing the same thing.